jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1466,6 +1500,46 @@ public String changeSessionId()
         return session.getId();
     }
 
+    /**
+     * Called when the request is fully finished being handled.
+     * For every session in any context that the session has
+     * accessed, ensure that the session is completed.
+     */
+    public void onCompleted()
+    {
+        if (_sessions != null && _sessions.size() > 0)
```

Comment:
No need to test for `_session.size > 0`, the iterator below will be easily eliminated by the JIT.